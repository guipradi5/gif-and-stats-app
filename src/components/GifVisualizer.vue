<template>
  <div class="gif-visualizer">
    <div class="gif-visualizer__wrapper">
      <img :src="element.media[0].gif.url" />
    </div>
    <div class="gif-visualizer__footer">
      <h3 class="gif-visualizer__description">
        {{ element.content_description }}
      </h3>
      <div class="gif-visualizer__data">
        <div class="gif-visualizer__data-size">
          <img src="@/assets/svg/file.svg" />
          {{ bytesToMb(element.media[0].gif.size) }} Mb
        </div>
        <div v-if="element.clickCount" class="gif-visualizer__data-clicks">
          <img src="@/assets/svg/data.svg" /> {{ element.clickCount }}
        </div>
      </div>
    </div>
  </div>
</template>

<script setup>
import { ref, computed, defineProps, onMounted } from "vue";

const props = defineProps(["element"]);
const element = computed(() => props.element);
const clickCount = ref(null);

function bytesToMb(bytes) {
  return (bytes / 1024 / 1000).toFixed(2);
}
</script>

<script>
export default {
  name: "GifVisualizer",
};
</script>

<style scoped lang="scss">
.gif-visualizer {
  box-shadow: $shadow;
  &__wrapper {
    background: rgb(51, 51, 51);
    display: flex;
    justify-content: center;
    border-top-left-radius: 10px;
    border-top-right-radius: 10px;
  }

  &__footer {
    border-bottom-left-radius: 10px;
    border-bottom-right-radius: 10px;
    padding: 20px;
    background: rgb(51, 53, 61);

    h3 {
      margin-top: 0;
    }
  }

  &__data {
    display: flex;

    > div {
      display: flex;
      margin-right: 20px;

      img {
        height: 1rem;
        width: 1rem;
        filter: invert(1);
        margin-right: 5px;
      }
    }
  }
}
</style>
