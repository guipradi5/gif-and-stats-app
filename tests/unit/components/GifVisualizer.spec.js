import { shallowMount } from "@vue/test-utils";
import GifVisualizer from "@/components/GifVisualizer.vue";

const defaultProps = {
  element: {
    clickCount: 5,
    media: [{ gif: { url: "https://gif.com/gif", size: 34564 } }],
  },
};
const component = (props) => {
  return shallowMount(GifVisualizer, {
    props: props || defaultProps,
  });
};
describe("GifVisualizer.vue", () => {
  it("GifVisualizer should mount", () => {
    const wrapper = component();
    expect(wrapper.vm).toBeTruthy();
  });

  it("GifVisualizer should render", async () => {
    const wrapper = await component();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
