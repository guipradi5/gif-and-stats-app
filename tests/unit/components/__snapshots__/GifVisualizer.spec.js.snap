// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifVisualizer.vue GifVisualizer should render 1`] = `
<div class="gif-visualizer">
  <div class="gif-visualizer__wrapper"><img src="https://gif.com/gif"></div>
  <div class="gif-visualizer__footer">
    <h3 class="gif-visualizer__description"></h3>
    <div class="gif-visualizer__data">
      <div class="gif-visualizer__data-size"><img src=""> 0.03 Mb </div>
      <div class="gif-visualizer__data-clicks"><img src=""> 5</div>
    </div>
  </div>
</div>
`;
